File: Daily Exercise/移除链表元素.py
class ListNode:
    def __init__(self,val = 0,next = None):
        self.val = val
        self.next = next

    def pNode(self,node = None):
        if node == None:
            return
        print(node.val)
        self.pNode(node.next)

class Solution:
    def removeElements(self, head:ListNode, val: int) -> ListNode:
        if head == None:
            return head
        head.next = self.removeElements(head.next,val)
        return head.next if (head.val == val) else head

    def __init__(self):
        node = ListNode(2)
        node.next = ListNode(3)
        node.next.next = ListNode(4)
        node.next.next.next = ListNode(3)
        node = self.removeElements(node,3)
        node.pNode(node)

if __name__ == "__main__":
    s = Solution()
    

File: Daily Exercise/移除链表元素.cpp
struct ListNode
{
    int val;
    ListNode *next;
    ListNode(): val(0), next(nullptr) {}
    ListNode(int x) : val(x),next(nullptr) {}
    ListNode(int x, ListNode *next): val(x), next(next){}

};


class Solution
{
private:
    /* data */
public:
    Solution(/* args */);
    ~Solution();

    ListNode* removeElements(ListNode* head, int val) {
        struct ListNode* dummyHead = new ListNode(0, head);
        struct ListNode* temp = dummyHead;
        while (temp->next != nullptr) {
        {
            if(temp->next->val == val) {
                temp->next = temp->next->next;
            }else {
                temp = temp->next;
            }
        }
        return dummyHead->next;
    }
};

Solution::Solution(/* args */)
{

}

Solution::~Solution()
{

}


